pkg/apis/cosigned/v1alpha1/clusterimagepolicy_types.go

```diff
@@ -49,9 +36,79 @@ func (*ClusterImagePolicy) GetGroupVersionKind() schema.GroupVersionKind {
 	return SchemeGroupVersion.WithKind("ClusterImagePolicy")
 }
 
-// ClusterImagePolicySpec holds the desired state of the ClusterImagePolicy (from the client).
+// ClusterImagePolicySpec defines a list of images that should be verified
 type ClusterImagePolicySpec struct {
-	// TODO(#1417): Flesh out the specification from the API spec.
+	Images []ImagePattern `json:"images"`
+}
+
+// ImagePattern defines a pattern and its associated authorties
+// If multiple patterns match a particular image, then ALL of
+// those authorities must be satisfied for the image to be admitted.
+type ImagePattern struct {
+	Glob        string      `json:"glob"`
+	Regex       string      `json:"regex"`
+	Authorities []Authority `json:"authorities"`
+}
+
+// The authorities block defines the rules for discovering and
+// validating signatures.  Signatures are
+// cryptographically verified using one of the "key" or "keyless"
+// fields.
+// When multiple authorities are specified, any of them may be used
+// to source the valid signature we are looking for to admit an
+// image.
+
+type Authority struct {
+	Key     KeyRef     `json:"key"`
+	Keyless KeylessRef `json:"keyless"`
+	Sources []Source   `json:"source"`
+	CTLog   TLog       `json:"ctlog"`
+}
+
+// This references a public verification key stored in
+// a secret in the cosign-system namespace.
+type KeyRef struct {
+	SecretRef *v1.SecretReference `json:"secretref"`
```

Comment:
because it's one of, I think all these should have:
// +optional

and add the json omitempty tag?